### PR TITLE
ZOMBIE PR!

### PR DESCRIPTION
**Pull request by andre-merzky**
Make ntp fully optional.
It is now only used when the ntp host resolves in the bootstrapper.
This commit also sneakily removes a prof.flush.

### CODE REVIEW

File: src/radical/pilot/utils/prof_utils.py [hunk 1]
```diff
@@ -115,15 +114,19 @@ def _timestamp_init(self):
         # We first try to contact a network time service for a timestamp, if that
         # fails we use the current system time.
         try:
-            import ntplib
-            ntphost  = os.environ.get('RADICAL_PILOT_NTPHOST', '0.pool.ntp.org')
-            response = ntplib.NTPClient().request(ntphost, timeout=1)
-            timestamp_sys  = response.orig_time
-            timestamp_abs  = response.tx_time
-            return [timestamp_sys, timestamp_abs, 'ntp']
+            ntphost = os.environ.get('RADICAL_PILOT_NTPHOST', '0.pool.ntp.org').strip()
```
Comment (marksantcroos) [inline src/radical/pilot/utils/prof_utils.py]: Doesn't the default value defeat the purpose?

File: src/radical/pilot/bootstrapper/default_bootstrapper.sh [hunk 2]
```diff
@@ -1309,7 +1309,7 @@ IFS=$OLD_IFS
 # we can't always lookup the ntp pool on compute nodes -- so do it once here,
 # and communicate the IP to the agent.  The agent may still not be able to
 # connect, but then a sensible timeout will kick in on ntplib.
-RADICAL_PILOT_NTPHOST=`dig +short 0.pool.ntp.org | grep -v "\.$" | head -n 1`
+RADICAL_PILOT_NTPHOST=`dig +short 0.pool.ntp.org | grep -v -e ";;" -e "\.$" | head -n 1`
```
Comment (marksantcroos) [inline src/radical/pilot/bootstrapper/default_bootstrapper.sh]: Good, this covers it afaics.

File: src/radical/pilot/utils/prof_utils.py (hunk 1 above)
Comment (andre-merzky) [inline src/radical/pilot/utils/prof_utils.py]: Ah, stupid me...
thanks!

**Comment by marksantcroos** [comment]
Tested on ARCHER and BW.

**Pull request merged**